torch/csrc/dynamo/eval_frame.c

```diff
@@ -478,14 +476,16 @@ static PyObject* call_guard_fail_hook(
     PyObject* hook,
     CacheEntry* e,
     size_t index,
-    PyObject* f_locals) {
+    PyObject* f_locals,
+    PyObject* f_globals) {
   // call debugging logic when a guard fails
   PyObject* args = PyTuple_Pack(
-      5,
+      6,
       e->check_fn,
       e->code,
-      f_locals,
       PyLong_FromSize_t(index),
+      f_locals,
+      f_globals,
```

Comment:
We shouldn't need to pass `f_globals` since it is known when we are creating the gaurds.  You can just bake it into a closure similar to what was done before.

Calling this function with anything other than the compile-time global scope wouldn't make sense.

Reply:
I asked the same question to Voz and he responded in personal chat. I'll let Voz recap the arguments here though...